### DISCUSSION TITLE
Network performance issues when uploading from Android to Server on WiFi / LAN

### DISCUSSION THREAD

**Discussion by mathwizmf2009** [immich-app/immich#4876] [Q&A]
I've been noticing an interesting issue regarding backups / uploads from my phone (Z Fold4) on WiFi to my Immich Server on LAN (within TrueNAS Scale).

Basically, when uploading a lot of photos (I'm still trying to perform the initial sync), my phone's network performance goes to crap. I can't use other apps requiring internet reliably. Worse than that, other devices on the same Access Point (U6-LR and U6-Pro) begin experiencing issues. Security cameras with borderline moderate SNR begin dropping from the network, streaming Emby from another server on LAN to my ShieldTV Pro on WiFi stops working, etc. It's like the app is being so demanding on the network and requiring so much bandwidth that other devices can't stay connected.  I've tested with my phone close to the AP, used a different AP, etc. As soon as I stop the backup, all network issues resolve - both internal to the phone and WiFi devices.

Before trying Immich, I've used NextCloud and did NOT experience anything like this when uploading mass amounts of photos. 

Has anyone else experienced this?

**Comment by alextran1502** [comment]
It sounds like you have a fast disk IO read time, which saturates your access point bandwidth. 

Which stats can you collect on the bandwidth usage of the access point?

**Reply by mathwizmf2009** [reply]
I've tried to look at the real-time bandwidth usage of the client (my phone) previously, but I questioned if the results were correct. They may have just been listing values to the WAN instead of including my LAN-LAN bandwidth. I'll need to dig into it a bit more. 

Do you happen to have any suggestions on how to get around this? If I manually transfer to a USB drive, import the files either via CLI, and then run the backup again, will it detect the duplicates and skip uploading them?

P.S. I love the work you're doing with this! It's been a lot of fun implementing all of this, and you have a lot of passionate community members implementing additional tools for things like easy Google Takeout imports via [Immich-go](https://github.com/simulot/immich-go). This is a blast and I think will have much better Wife Approval Factor than NextCloud! lol

**Comment by TheChilledBuffalo** [comment]
Hey @mathwizmf2009 ! I know its been a while since you posted this, but I'm facing a very similar if not the exact same issue. Did you come up with a solution or a workaround?

**Reply by mathwizmf2009** [reply]
It's been a while, but if I remember correctly I ended up using a usb-eth adapter and plugged in my phone direct to ethernet.

My wife's budget phone did not create this problem on my network, so I think Alex was correct.

I'm also not sure if the duplication detection that I asked about has had any changes since then. Previously the duplication detection happened 100% on the server, meaning everything still needed to be uploaded from phone to server prior to being determined if it existed - which made the usb transfer option non-viable at the time. I haven't been tracking release notes lately though.

**Reply by TheChilledBuffalo** [reply]
Thank you! A wired local connection worked!
